pyqtorch/circuit.py

```diff
@@ -59,18 +61,79 @@ def run(self, state: State = None, values: dict[str, Tensor] | ParameterDict = {
         return self.forward(state, values)
 
     def forward(self, state: State, values: dict[str, Tensor] | ParameterDict = {}) -> State:
+        if self.dropout:
+            mode = self.dropout["mode"]
+            assert mode is str
+            dropout_mode = self.select_dropout_mode(mode)
+            state = dropout_mode(state, values)
+
+        else:
+            for op in self.operations:
+                state = op(state, values)
+
+        return state
+
+    def select_dropout_mode(self, mode: str) -> Callable:
+        modes = {
+            "rotational": self.rotational_dropout,
```

Comment:
@callumfduffy we can use a `StrEnum` for this like for the `DiffMode` in pyqtorch.utils